test/include/seqan3/test/expect_range_eq.hpp

```diff
@@ -0,0 +1,54 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides test utilities for std::ranges::range types.
+ * \author Marcel Ehrhardt <marcel.ehrhardt AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <seqan3/std/algorithm>
+#include <seqan3/std/ranges>
+
+#include <seqan3/core/detail/debug_stream_range.hpp>
+#include <seqan3/core/detail/to_string.hpp>
+#include <seqan3/test/pretty_printing.hpp>
+
+namespace seqan3::test
+{
+#define EXPECT_RANGE_EQ(val1, val2) \
+    EXPECT_PRED_FORMAT2(::seqan3::test::expect_range_eq{}, val1, val2);
+
+struct expect_range_eq
+{
+    template <typename rng_t>
+    auto copy_range(rng_t && rng)
+    {
+        using value_t = std::ranges::range_value_t<rng_t>;
+        std::vector<value_t> rng_copy{};
+        std::ranges::copy(rng, std::ranges::back_inserter(rng_copy));
+        return rng_copy;
+    }
+
+    template <std::ranges::range lhs_t, std::ranges::range rhs_t>
```

Comment:
what is the compiler message if the range value types are not comparable? Should this be caught here?

Reply:
I don't know and I'm unfortunately short on time to test this right now, I think you would get some compiler error that 

```c++
if (std::ranges::equal(lhs_copy, rhs_copy))
```

can't be compiled. I expect that you will get a nice stack trace with the involved types and you will see that `"if (std::ranges::equal(lhs_copy, rhs_copy))"` failed for your given ranges. My guess is that this will be less cryptic than putting that requirement on a random `operator()`.